tests/water_movement_test.cpp

```diff
@@ -144,7 +144,7 @@ TEST_CASE( "avatar_diving", "[diving]" )
 
     // Put us back at 0. We shouldn't have to do this but other tests are
     // making assumptions about what z-level they're on.
-    g->vertical_shift( 0 );
+    dummy.setpos( test_origin );
```

Comment:
The issue comes from this change. `game::vertical_shift` sets `abs_sub` on the `map` object. And apparently `Creature::setpos` does not, so the map still thinks it's at z-level -1 when the avatar moves in subsequent tests, which sets the avatar z-level to -1 when `game::update_map` is called.